test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/GetRestoreSettingTaskTests.cs

```diff
@@ -371,5 +407,19 @@ public void GetRestoreSettingsTask_VerifyEmptyPerFrameworkSettings()
                   <add key=""outer-key"" value=""outer-value"" />
                 </SectionName>
               </configuration>";
+
+        private static string DisableSourceConfig =
+            @"<?xml version=""1.0"" encoding=""utf-8""?>
+             <configuration>
+              <packageSources>
+                <Clear/>
+                <add key=""NuGet.org"" value=""https://api.nuget.org/v3/index.json"" />
+                <add key=""NuGet.org.v2"" value=""https://nuget.org/v2/api"" />
+              </packageSources>
+              <disabledPackageSources>
+                 <add key=""NuGet.org"" value=""true"" />
```

Comment:
you can add a clear above disabled also I think. More unique names for these keys could help too since this will get combined with the machine's settings.